### ISSUE TITLE
Unexpected behaviour: Hiding console after maximizing window reverts back to old size.

### ISSUE THREAD

**Issue by webbertakken** [flyingpie/windows-terminal-quake#105]
**Context**

This is the default size of my console:

![image](https://user-images.githubusercontent.com/20756439/141689247-da0afaeb-b493-497a-afe2-152760466e3e.png)

When clicking on the maximize window icon - which can sometimes be helpful when having to fit more on the screen - the console looks like this:

![image](https://user-images.githubusercontent.com/20756439/141689286-b6074793-5d32-4ea8-a748-6434b490a223.png)

This is a pretty great default behaviour that works as expected.

**Problem**

However when I now hide or re-show the terminal using the quake hotkey the windows size reverts to the one from the initial settings.

**Suggested solution**

I would suggest the following behaviour:

- When clicking "Maximise", update the current size in quake console.
- When clicking "Restore down", restore to the size from the configuration file.

**Comment by flyingpie** [comment]
Couple notes:
- How would we be able to notice the window size having changed _due to a user action_? Window sizes can change for different reasons (I've used toggling myself exactly to restore the original size)
- When using multiple monitors with different sizes, the relative sizing we do now handles that pretty neatly. When we want to maintain the user-defined size, what do we do then? Calculate relative size? Only apply when on the same monitor?
- There is some overlap with the "detach" feature being discussed here:
  https://github.com/flyingpie/windows-terminal-quake/issues/184

**Comment by sujal-wpc** [comment]
I've managed to achieve this behavior by setting `Position > Resize app window` to `Never`. 
Config file version would be something like:: 

```jsonc
{
	"$schema": "wtq.schema.json",
	"Apps": [
		{
			"Name": "Windows Terminal",
			"Resize": "Never",
		}
	],
}
```

This does create an annoyance if you have multiple monitors with varying resolutions as you might need to resize them manually whenever you spawn the terminal on a different screen.

@flyingpie would it be possible to add a `None` option for `HorizontalAlign`? I want to be able to move the quake terminal around, which can be done, but then toggling the terminal resets the alignment based on this setting.

**Comment by flyingpie** [comment]
@sujal-wpc Thank you for raising this point!

The resizing part has indeed been added since this issue opened, so I should update the work that's left.

Regarding your question directly, could you elaborate a bit on the use case? You don't want to resize the app window ever, or is this more of a workaround that comes close enough?

**Comment by sujal-wpc** [comment]
@flyingpie Quite the contrary, actually. I want to resize the window, which works well. Just that if I move the window around, toggling the terminal will reset its window position due to the horizontal alignment setting.

A common usecase for this is just resizing the terminal manually to about half the screen width, then moving it to the left, and having another window to the right (for e.g: browser with docs open ). When toggling the terminal, it then moves back to the center, thus obstructing the other window.

**Comment by flyingpie** [comment]
@sujal-wpc Yeah, sorry, I meant "you don't want _WTQ_ to resize the window, ever".

If I'm understanding correctly, WTQ should not mess with the window's size or horizontal position at all, and just move it off- and on the screen, correct?

Do you have thoughts on how this should work with multiple monitors, especially if they're different resolutions (i.e., you want a different sizing on the monitor)?

**Comment by sujal-wpc** [comment]
Ah ok. I'm fine with having settings that enable / disable if WTQ can control the size (already has setting for this) and control the alignment (just needs a None option).

I am fine with the current behaviour tbh since I am using monitors with the same resolution now. But maybe one solution would be to have WTQ remember percentage size relative to the current resolution (and maybe also remember percentage position from left, top) and set accordingly on other monitors?

Though for now, I'm fine with just having the None option for horizontal alignment

**Comment by flyingpie** [comment]
Good to know that your use case would be handled with just the disabling of alignment, let's scope to that for now. Since resize already works, I'm not expecting this to be too hard, I'll get back to this.

**Comment by flyingpie** [comment]
@sujal-wpc I've pushed a prerelease that includes this:
https://github.com/flyingpie/windows-terminal-quake/releases/tag/vHorizontalAlignNone

Basically just what you suggested, a "HorizontalAlign": "None":

```jsonc
{
	"Apps": [
		{
			"Name": "Dolphin",
			"Hotkeys": [{ "Modifiers": "Control", "KeyChar": "2" }],
			"FileName": "dolphin",
			"HorizontalAlign": "None"
		}
	]
}
```

Now, there's a conflict with the background watcher that keeps the window where it should. This feature exists to deal with monitor (dis)connects, resolution changes, that sorta thing.

Whenever a window moves or is resized, it kicks the window back to where it "should" be, though the "should" has changed with this new setting.

I've disabled that entirely in this build, so we can at least test the feature itself.

Wanna check a look? :)

**Comment by sujal-wpc** [comment]
Thanks. The "no horizontal align" persists properly. 
I do miss not being able to toggle it on different monitors now :p

**Comment by flyingpie** [comment]
Yes, I've noticed that too in my testing, but I think we can take that into account as well. There have been some more requested features to do with what monitor things are at.

Thank you for testing the current release, I'll notify you for the next update!
